src/Xrefcheck/Verify.hs

```diff
@@ -506,17 +506,17 @@ verifyReference
       -> IO (Either VerifyError ())
     retryVerification rc resIO = do
       res <- resIO
-      case res of
+      case (rInfo, res) of
         -- Success
-        Right () -> do
-          modifyProgressRef Nothing $ reportSuccess rLink
+        (RIExternal (ELUrl url), Right ()) -> do
+          modifyExternalProgress Nothing $ reportSuccess url
           pure res
-        Left err -> do
+        (RIExternal (ELUrl url), Left err) -> do
```

Comment:
Hm, such complex pattern match that checks `2x2` cases + one of the cases is a nested thing - seems dangerous to me. If one day we need to treat another constructor as external link, bugs or code bloat will likely be introduced.

I would handle the difference between `()` and `Text` witnesses in some other way.

Maybe leave `modifyProgressRef`, but make it accept `(w -> Progress Int w -> Progress Int w)` function? As before, let in pattern match on `rInfo` inside but also make it supply `()` / `url` argument, not expect its caller to do so?

Reply:
Good point. It can be kept as before by using a `(forall w. w -> Progress Int w -> Progress Int w)` function.

Reply:
While maybe functions like `getTaskTimestamp` can really not accept a witness, I don't see what will break in this case.

Reply:
I made `getTaskTimestamp` and `setTaskTimestamp` accept a witness in #265 because, from the interface point of view, they seem like operations that require a witness, but the current implementation does not really need a witness.

Do you prefer to make them not require a witness? I'm also not completely sure about what this progress bar timestamp means.

Reply:
That time is displayed when rate limiteters asked us to wait.

I'd say that following the interface sanity is really important, however the description of `TaskTimestamp` and `pTaskTimestamp` say that those are timestamps of an anonymous timer task, so actually the semantics also assumes here that the witness is irrelevant, as far as I see it.

Reply:
Oh, I missed the "anonymous" part. I agree then with the irrelevance of the witness in these functions.

Maybe this is overthinking: from the interface point of view, `getTaskTimestamp` surely does not require a witness because it retrieves a `TaskTimestamp`, which is anonymous, but `setTaskTimestamp` constructs a `TaskTimestamp`, so maybe it can be seen as a point in where a task passes to be anonymous, and then it would be justified to require a witness there.

Reply:
Actually, by following the code, I think that this timestamp is just informative and has nothing to do with progress units, right? I will change both functions then.

Reply:
> so maybe it can be seen as a point in where a task passes to be anonymous (such a task should exist in order to perform the action), and then it would be justified to require a witness there.

Hmm, actually were it me who is doing this task, and if I wanted to make it clear and neat, I guess I would think in the same manner. And that timestamp is semantically tightly coupled with fixable failures, only they can contribute to that timer.

What made me dislike the need in witness was primarily the fact that sometimes we pass dummy `""` as a witness to those functions. But given your reasoning, and given this `""` occurs only in tests, that seems justified.

So I got convinced. Fell free to revert that commit; probably it would also make sense to provide your reasoning in haddock, otherwise I guess someone else will just remove that witness argument in the future.